File: Data Structures & Algorithms/binary-search/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int L = 0, R = nums.length - 1, M = 0;

        while (L <= R) {
            M = (L + R) / 2;

            if (target > nums[M]) {
                L = M + 1;
            } else if (target < nums[M]) {
                R = M - 1;
            } else {
                return M;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    getConcatenation(nums) {
        const arr = new Array(nums.length + nums.length);
        let count = 0;
        for (let i = 0; i < nums.length; i++) {
            arr[count] = nums[i];
            count++;
        }
        for (let i = 0; i < nums.length; i++) {
            arr[count] = nums[i];
            count++;
        }
        return arr;
    }
}
